app/queries/get_expanded_links.rb

```diff
@@ -1,42 +1,75 @@
 module Queries
   class GetExpandedLinks
-    def self.call(content_id, locale, with_drafts: true)
-      if (link_set = LinkSet.find_by(content_id: content_id))
-        expanded_link_set(link_set, locale, with_drafts: with_drafts)
-      elsif Document.where(content_id: content_id).exists?
-        empty_expanded_link_set(content_id)
-      else
-        error_details = {
-          error: {
-            code: 404,
-            message: "Could not find link set with content_id: #{content_id}"
-          }
-        }
+    def self.call(content_id, locale, with_drafts: true, generate: false)
+      self.new(content_id, locale, with_drafts, generate).call
+    end
 
-        raise CommandError.new(code: 404, error_details: error_details)
+    def call
+      pre_computed = generate ? nil : pre_computed_link_set
```

Comment:
Would it be simpler to use a naming convention like `generate_links`, `generate_links_response` and `generated_links`, `generated_links_response` (or even `pre_generated_links` etc.)? Not intended as a PR blocker.